design-docs/2021/q1/credrank-migration.md

```diff
@@ -0,0 +1,97 @@
+# CredRank Migration Design Doc
+
+## Last Updated: 2021.01.07
+
+## Authors:
+ - @decentralion
+
+
+## Overview
+
+We've finished implementing [CredRank], but we haven't yet migrated SourceCred to use it.
+This doc outlines a plan for migrating all of SourceCred to use Cred scores from CredRank,
+and to remove [Timeline Cred].
+
+[CredRank]: https://discourse.sourcecred.io/t/credrank-scalable-interpretable-flexible-attribution/654
+[Timeline Cred]: https://github.com/sourcecred/sourcecred/pull/1212
+
+## Steps
+
+### Finish Implementing CredGrainView
+
+The `CredGrainView` (see src/core/credGrainView.js) is a class that makes it
+convenient to retrieve Cred (and Grain) data about any participant. It will act
+as the main data API layer for consumers that need access to Cred data. It's
+thus a replacement for the existing `CredView` class at
+src/analysis/credView.js.
+
+Currently, CGV exists as an interface, but most of the methods are not
+implemented and just throw errors. It needs to be implemented and tested.
+
+### Create a CredRank-compatible Cred Explorer
+
+Presently, the Explorer (src/ui/components/Explorer/Explorer.js) depends on the
+`CredView` class, which means it uses TimelineCred scores. We should create a
+new overview page that depends on `CredGrainView` instead.
+
+Less ambitiously, we could just port the existing UI over to use `CredView`.
```

Comment:
Worth noting that this will likely regress the top contributions dropdown feature.

Reply:
Done. @topocount I think when we do the migration, we will regress the top contributions dropdown until we do a UI redesign based on @KuraFire's work. And I think that's OK b.c. the current feature is nearly useless due to its lack of context.

Reply:
+1, I would even be fine if that feature is simply taken out until the new UI redesign for Participants view is done. 😂